src/shared/rpc/methods/sign-psbt.ts

```diff
@@ -3,44 +3,35 @@ import * as btc from '@scure/btc-signer';
 import * as yup from 'yup';
 
 import { networkModes } from '@shared/constants';
-import { isNumber, isUndefined } from '@shared/utils';
 
-function testIsNumberOrArrayOfNumbers(value: unknown) {
-  if (isUndefined(value)) return true;
-  if (Array.isArray(value)) return value.every(item => isNumber(item));
-  return isNumber(value);
-}
+import {
+  accountSchema,
+  formatValidationErrors,
+  getRpcParamErrors,
+  testIsNumberOrArrayOfNumbers,
+  validateRpcParams,
+} from './validation.utils';
 
-const rpcSignPsbtValidator = yup.object().shape({
-  publicKey: yup.string().required(),
+const rpcSignPsbtParamsSchema = yup.object().shape({
+  account: accountSchema,
   allowedSighash: yup.array().of(yup.mixed().oneOf(Object.values(btc.SignatureHash))),
   hex: yup.string().required(),
-  signAtIndex: yup.mixed<number | number[]>().test(testIsNumberOrArrayOfNumbers),
   network: yup.string().oneOf(networkModes),
```

Comment:
Can we adjust network modes to be all the bitcoin modes, and not just mainnet/testnet? We had report of signet not working and figure this is why